integration-tests/smoke/ccip/ccip_rmn_test.go

```diff
@@ -448,9 +404,241 @@ type rmnTestCase struct {
+			rmnN := rmnCluster.Nodes["rmn_"+strconv.Itoa(n.id)]
+			require.NoError(t, osutil.ExecCmd(zerolog.Nop(), "docker kill "+rmnN.Proxy.ContainerName))
+			t.Logf("Paused RMN node %d", n.id)
+		}
+	}
+}
+
+func (tc rmnTestCase) disableOraclesIfThisIsACursingTestCase(ctx context.Context, t *testing.T, envWithRMN changeset.DeployedEnv) []string {
+	disabledNodes := make([]string, 0)
+
+	if len(tc.cursedSourceChainIdxs) > 0 || tc.globalCurse {
+		listNodesResp, err := envWithRMN.Env.Offchain.ListNodes(ctx, &node.ListNodesRequest{})
+		require.NoError(t, err)
+
+		for _, n := range listNodesResp.Nodes {
+			if strings.HasPrefix(n.Name, "bootstrap") {
+				continue
+			}
+			_, err := envWithRMN.Env.Offchain.DisableNode(ctx, &node.DisableNodeRequest{Id: n.Id})
+			require.NoError(t, err)
+			disabledNodes = append(disabledNodes, n.Id)
+			t.Logf("node %s disabled", n.Id)
+		}
+	}
+
+	return disabledNodes
+}
+
+func (tc rmnTestCase) sendMessages(t *testing.T, onChainState changeset.CCIPOnChainState, envWithRMN changeset.DeployedEnv) (map[uint64]*uint64, map[changeset.SourceDestPair]uint64, map[changeset.SourceDestPair][]uint64) {
+	startBlocks := make(map[uint64]*uint64)
+	seqNumCommit := make(map[changeset.SourceDestPair]uint64)
+	seqNumExec := make(map[changeset.SourceDestPair][]uint64)
+
+	for _, msg := range tc.messagesToSend {
+		fromChain := tc.pf.chainSelectors[msg.fromChainIdx]
+		toChain := tc.pf.chainSelectors[msg.toChainIdx]
+
+		for i := 0; i < msg.count; i++ {
+			msgSentEvent := changeset.TestSendRequest(t, envWithRMN.Env, onChainState, fromChain, toChain, false, router.ClientEVM2AnyMessage{
+				Receiver:     common.LeftPadBytes(onChainState.Chains[toChain].Receiver.Address().Bytes(), 32),
+				Data:         []byte("hello world"),
+				TokenAmounts: nil,
+				FeeToken:     common.HexToAddress("0x0"),
+				ExtraArgs:    nil,
+			})
+			seqNumCommit[changeset.SourceDestPair{
+				SourceChainSelector: fromChain,
+				DestChainSelector:   toChain,
+			}] = msgSentEvent.SequenceNumber
+			seqNumExec[changeset.SourceDestPair{
+				SourceChainSelector: fromChain,
+				DestChainSelector:   toChain,
+			}] = []uint64{msgSentEvent.SequenceNumber}
+			t.Logf("Sent message from chain %d to chain %d with seqNum %d", fromChain, toChain, msgSentEvent.SequenceNumber)
+		}
+
+		zero := uint64(0)
+		startBlocks[toChain] = &zero
+	}
+
+	return startBlocks, seqNumCommit, seqNumExec
+}
+
+func (tc rmnTestCase) callContractsToCurseChains(ctx context.Context, t *testing.T, onChainState changeset.CCIPOnChainState, envWithRMN changeset.DeployedEnv) {
+	for _, remoteCfg := range tc.remoteChainsConfig {
+		remoteSel := tc.pf.chainSelectors[remoteCfg.chainIdx]
+		chState, ok := onChainState.Chains[remoteSel]
+		require.True(t, ok)
+		chain, ok := envWithRMN.Env.Chains[remoteSel]
+		require.True(t, ok)
+		for _, chainSel := range tc.pf.cursedSourceChains.ToSlice() {
+			txCurse, errCurse := chState.RMNRemote.Curse(chain.DeployerKey, chainSelectorToBytes16(chainSel))
```

Comment:
Maybe check that `chainSel != remoteSel` because we don't curse the chain selector subject if the chain that we are cursing has the same selector as that subject.

i.e if you are cursing `A`, on all chains other than `A`, you `Curse(subject(A))`, but on `A` itself, you curse with `Curse(GLOBAL_CURSE_SUBJECT)`. At least this is my understanding, worth confirming.